scripts/test-gem-build

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env bash
+#
+#  run as part of CI
+#
+if [[ $# -lt 2 ]] ; then
+  echo "usage: $(basename $0) <output_dir> <platform>"
+  exit 1
+fi
+
+set -e -u
+
+OUTPUT_DIR=$1
+BUILD_NATIVE_GEM=$2
+
+test -e /etc/os-release && cat /etc/os-release
+
+set -x
+
+bundle install --local || bundle install
+bundle exec rake set-version-to-timestamp
+
+if [[ "${BUILD_NATIVE_GEM}" == "ruby" ]] ; then
+  # TODO we're only compiling so that we retrieve tarballs, we can do better.
```

Comment:
Is there a way to download MiniPortile dependencies without also compiling the gem?

Reply:
We could implement the download ourselves (such as https://github.com/stanhu/pg_query/blob/fc420181abd2b551d90bb91b240431cf52f85d82/ext/pg_query/extconf.rb#L18-L24) or call `MiniPortile.download` (https://github.com/flavorjones/mini_portile/blob/ee637e0a76e5ae7144949d868948b45d8d858051/lib/mini_portile2/mini_portile.rb#L108C3-L112).

Reply:
Do you think that’d be worth doing as a separate Rake task for the bare, uncompiled Ruby gem to be speed up the build?

Reply:
Yes. https://github.com/mudge/re2/actions/runs/5882973962/job/15954634773 took 2-3 minutes to download and compile, but we could probably make that finish in less than 30 seconds.

Reply:
#89 makes it so that this job now finishes in 17 seconds (https://github.com/mudge/re2/actions/runs/6017912200/job/16325121343?pr=89).